Automattic/wp-calypso
# Components: Use ToggleControl in our FormToggle component

This PR updates our `FormToggle` component to use `ToggleControl` from `@wordpress/components`. This affects all of Calypso's form toggle usages, so it will require thorough testing of all affected toggles.

We're altering the Calypso form toggle component as this makes the diff smaller and easier to test. This results in the Calypso component being a wrapper for the `@wordpress/components` form toggle control component. Eventually, if this wrapper is not necessary, we can remove it in another PR and replace with direct `ToggleControl` usage.

#### Changes proposed in this Pull Request

* Replace the Calypso toggle with a `@wordpress/components` toggle everywhere, fix tests and remove obsolete styles

To showcase how expected differences look like, I'm attaching a before / after of a card in the Discussion settings screen.

Before:
![](https://cldup.com/gDDDucOpjP.png)

After:
![](https://cldup.com/IdMQ7aIj1e.png)

#### Testing instructions

Note: There will be visual differences because the old Calypso toggle was more narrow. So in testing, you'd basically want to make sure that the introduced differences make sense and look well. 

* Go through the following instances with toggles and verify the toggles look well and work the same way they do on production:
  * `/devdocs/blocks/post-share` - all toggles in the example
  * `/following/manage` - the toggles in the Settings popover of a site you're following
  * `/settings/taxonomies/category/:site` - click on "Add New Category" and observe the "Top level category" toggle
  * `/devdocs/design/form-fields` - the toggle example
  * `/devdocs/design/g-suite` - toggles at the bottom of the example
  * `/devdocs/playground` - try inserting a `<FormToggle />` in the DevDocs playground and you should see a working compact toggle.
  * `/store/product/:siteSlug/:productId`, where `:siteSlug` is the site slug of a site with WooCommerce, and `:productId` is the ID of a product, observe the "product with variations..." and "Promote this product across the store" toggles.
  * `/store/settings/taxes/:siteSlug` where where `:siteSlug` is the site slug of a site with WooCommerce and WooCommerce Services active, and a valid USA address selected as the store address, observe the "Tax calculations enabled" toggle.
  * `/store/settings/shipping/:siteSlug` where where `:siteSlug` is the site slug of a site with WooCommerce and WooCommerce Services active, observe the toggle at the right of the "Shipping Labels" card heading.
  * `/store/settings/email/:siteSlug` where where `:siteSlug` is the site slug of a site with WooCommerce, WooCommerce Services, and Mailchimp for WooCommerce active, observe all the toggles on the page.
  * `/store/settings/payments/:siteSlug` where where `:siteSlug` is the site slug of a site with WooCommerce and WooCommerce Services, observe the toggles next to each of the available payment methods.
  * `/store/settings/shipping/:siteSlug` where where `:siteSlug` is the site slug of a site with WooCommerce and WooCommerce Services, observe the toggles in the modal when editing a shipping method, in the list of shipping methods, and in the shipping labels section.
  * `/extensions/wp-super-cache/:siteSlug` where `:siteSlug` is a site slug of a Jetpack site with WP Super Cache installed. Go through all tabs and look at all toggles.
  * `/me/privacy` - the toggle at the bottom of the "Usage Information" card.
  * `/me/purchases` - select a purchase, and observe the toggle in the Subscription Renewal section (the rest of the auto-renew toggle instances in the domain management section were already compact ones).
  * `/domains/manage/:siteSlug/contacts-privacy/:domain` where `:siteSlug` your site and `:domain` is a domain on that site, observe the "Privacy Protection" toggle
  * `/domains/manage/:siteSlug/dns/:domain` where `:siteSlug` is your site and `:domain` is a domain on that site, observe the domain connect toggle. It's possible it can be disabled if the domain is managed by WP.com currently
  * `/domains/manage/:siteSlug/name-servers/:domain` where `:siteSlug` your site and `:domain` is a domain on that site, observe the "Use WordPress.com Name Servers" toggle
  * `/earn/ads-settings/:siteSlug` where `:siteSlug` is the slug of a site that has WordAds enabled. Observe the multiple toggles on this page
  * `/earn/payments-plans/:site` where `:site` is the slug of a simple WP.com site with a business plan. You will have to authorize Stripe first before accessing that screen. When there, click on the "Add a new payment plan" button and observe the two toggles in the modal.
  * `/plugins/manage/:siteSlug` where `:siteSlug` is the slug of a Jetpack site with plugins. Observe the toggles on each plugin.
  * `/settings/discussion/:siteSlug` - where `:siteSlug` is the slug of a site; go through all the settings sections for both a WP.com site and Jetpack site (ideally both with high tier paid plans) and observe the toggles.
  * `/marketing/traffic/:siteSlug` - where `:siteSlug` is the slug of a site; there are multiple toggles on that page (especially if you're on a high-tier paid plan). 
  * `/settings/performance/:site?tour=jetpackSiteAccelerator` where `:site` is a Jetpack site slug with a paid plan, and verify the guided tour still initiates and works properly, pointing to the toggle of the site accelerator.
* Verify all tests pass.
